miniapps/tools/nodal_transfer.cpp

```diff
@@ -0,0 +1,257 @@
+// Copyright (c) 2010-2023, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details
+
+#include <mfem.hpp>
+#include <fstream>
+#include <iostream>
+#include <cmath>
+#include "../common/mfem-common.hpp"
+
+/// Generate second order mesh on 4 processes
+/// mpirun -np 4 ./nodal_transfer -rs 2 -rp 1 -gd 1 -o 2
+/// Read the generated data and map it to a grid function
+/// defined on two processes
+/// mpirun -np 2 ./nodal_transfer -rs 2 -rp 0 -gd 0 -snp 4 -o 2
+///
+/// Generate first order grid function on 8 processes
+/// mpirun -np 8 ./nodal_transfer -rs 2 -rp 2 -gd 1 -o 1 -m ../../data/star.mesh
+/// Read the generated data on 4 processes and coarser mesh
+/// mpirun -np 4 ./nodal_transfer -rs 2 -rp 0 -gd 0 -snp 8 -o 1 -m ../../data/star.mesh
```

Comment:
Please move this above and include a banner in the same style as the other miniapps e.g. `lor-transfer.cpp`

Reply:
Do we want to highlight this miniapp in `../doc/CodeDocumentation.dox`?

Reply:
Miniapp documentation is added!

I will add the other documentation once  the rest of the features are implemented - parallel search,...